arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -250,18 +250,15 @@ Result RocksDBReplicationContext::getInventory(TRI_vocbase_t* vocbase,
 
 // iterates over at most 'limit' documents in the collection specified,
 // creating a new iterator if one does not exist for this collection
-RocksDBReplicationResult RocksDBReplicationContext::dump(
-    TRI_vocbase_t* vocbase, std::string const& collectionName,
+RocksDBReplicationResult RocksDBReplicationContext::dumpJson(
+    TRI_vocbase_t* vocbase, std::string const& cname,
     basics::StringBuffer& buff, uint64_t chunkSize) {
   TRI_ASSERT(_users > 0 && !_exclusive);
   CollectionIterator* collection{nullptr};
-  auto release = [&]() -> void {
-    if (collection) {
-      MUTEX_LOCKER(locker, _contextLock);
-      collection->release();
-    }
-  };
-  TRI_DEFER(release());
+  TRI_DEFER(if (collection) {
+    MUTEX_LOCKER(locker, _contextLock);
+    collection->release();
+  });
```

Comment:
Why un-lambda it?

Reply:
because TRI_DEFER also wraps it in a lambda

Reply:
...this is why I inherently mistrust macros. :(